### ISSUE TITLE
Remove "Thank You" Animation

### ISSUE THREAD

**Issue by DrakkenWolfgang** [ClaraCrazy/Flipper-Xtreme#119]
Hello. New to the Flipper scene. Just got mine a couple of days ago & it has been quite the experience! Have been enjoying tinkering around with it & learning more about it hands on.

I have a slight issue & hoping someone can point me in the right direction.

I have the latest release Xtreme Firmware installed & am curious to know if there is a way to remove "Thank You" splash animation & how would i go about it?

It seems to want to override the stock flipper animation at times. Sometimes a reboot shows the stock flipper animations for a while, then goes back to the "Thank You" animation.

Any info would be appreciated! ![20230130_190517.jpg](https://user-images.githubusercontent.com/84858901/215634143-7a802ff7-0486-49ab-a771-81fb3ed88de8.jpg)

![20230130_190519.jpg](https://user-images.githubusercontent.com/84858901/215634146-96d8c4c5-f0b0-40e9-9b2d-3e51b55a8983.jpg)

![20230130_190524.jpg](https://user-images.githubusercontent.com/84858901/215634144-7d00d8be-a28d-478c-a799-2612d1b038c7.jpg)

![20230130_190518(0).jpg](https://user-images.githubusercontent.com/84858901/215634141-4d2a5851-2bc5-407d-8fb3-75dbcca8e6d6.jpg)

![20230130_190522.jpg](https://user-images.githubusercontent.com/84858901/215634145-dd650027-4428-4557-998b-0c700930e557.jpg)

![20230130_190525.jpg](https://user-images.githubusercontent.com/84858901/215634149-f3364da3-8ac7-4f19-a5ab-b789edccd9dd.jpg)

![20230130_190518.jpg](https://user-images.githubusercontent.com/84858901/215634147-21f02695-82d0-4b89-b057-9d2fca61a7a2.jpg)

![20230130_190523.jpg](https://user-images.githubusercontent.com/84858901/215634140-cce8e3d1-cf9a-44ae-9fe3-73eb3b0b190b.jpg)

![20230130_190521.jpg](https://user-images.githubusercontent.com/84858901/215634142-2449bab2-6c7c-4aa7-a4a3-44e38c6bef42.jpg)

![20230130_190520.jpg](https://user-images.githubusercontent.com/84858901/215634148-c11b52d3-f8c8-464c-bee4-8bc8545da74e.jpg)

**Comment by Willy-JL** [comment]
The thank you is the basic fallback builtin animation for this firmware, like for the official firmware its the dolphin looking at the tv on the couch. On flipper the fallback animation is used in 3 cases:
1. No other suitable animation was found. For example you are not high level enough for any of the animations (would occur only on custom packs, the included animations are all scaled correctly with levels).
2. A suitable animation was found, but it failed loading. Could be because of too many frames, some incorrect metadata, or frame files being too big. Again, only an issue on custom asset packs since the pre-included animations were all tested.
3. It gets chosen randomly. When choosing an animation to play its picked randomly between the ones listed in the animation manifest, and the built in animations. So there is a chance that the fallback animation shows up. Depending on how many other animations are suitable, this is more or less likely to happen. If you are level 1, there are 2 possible animations: the level 1 animation (with weight 7), and the fallback animation (with weight 1), so there is a 1 in 8 chance (12%) you see the thankyou animation. If you are level 30, there are 30 suitable animations (all weight 7) plus the fallback one (at weight 1), aka about a 1 in 211 chance  (0.4%) you see the built in animation.

To summarize:
- Dont remove it, it is used as a fallback animation for when things go wrong
- Get more animations so you see it less often

**Comment by DrakkenWolfgang** [comment]
> The thank you is the basic fallback builtin animation for this firmware, like for the official firmware its the dolphin looking at the tv on the couch. On flipper the fallback animation is used in 3 cases:
> 1. No other suitable animation was found. For example you are not high level enough for any of the animations (would occur only on custom packs, the included animations are all scaled correctly with levels).
> 2. A suitable animation was found, but it failed loading. Could be because of too many frames, some incorrect metadata, or frame files being too big. Again, only an issue on custom asset packs since the pre-included animations were all tested.
> 3. It gets chosen randomly. When choosing an animation to play its picked randomly between the ones listed in the animation manifest, and the built in animations. So there is a chance that the fallback animation shows up. Depending on how many other animations are suitable, this is more or less likely to happen. If you are level 1, there are 2 possible animations: the level 1 animation (with weight 7), and the fallback animation (with weight 1), so there is a 1 in 8 chance (12%) you see the thankyou animation. If you are level 30, there are 30 suitable animations (all weight 7) plus the fallback one (at weight 1), aka about a 1 in 211 chance  (0.4%) you see the built in animation.
> 
> Also, with v40 we added a check to make sure the same animation is not played twice, so worse comes to worst youre level 1 and you see level 1 anim, then fallback anim, then lvl 1 anim, then fallback anim...
> 
> To summarize:
> - Dont remove it, it is used as a fallback animation for when things go wrong
> - Get more animations so you see it less often

Thanks for the response! Definitely clears some info I didn't understand. My next research point with Flipper will be custom animations/ icons. That is a long read in itself, but I'll figure it out eventually & maybe make some.🤣

**Comment by Willy-JL** [comment]
> Thanks for the response! Definitely clears some info I didn't understand. My next research point with Flipper will be custom animations/ icons. That is a long read in itself, but I'll figure it out eventually & maybe make some.rofl

Good luck with that, feel free to join the discord for some help. Also, if you just look online youll see the common and annoying way to do this, which is replacing existing animations and icons on the sd or the firmware. We have our own system with asset packs that takes away a good chuck of that struggle, you can learn about that [here](https://github.com/ClaraCrazy/Flipper-Xtreme/wiki/1.-File-Formats) (the bottom section with animations icons and packs)
